### ISSUE TITLE
Computing Q, K, and V in ViT

### ISSUE THREAD

**Issue by arthurfeeney** [HPCForge/Mondrian-DDOp#5]
ATM, I'm just using spectral convolutions for computing Q, K, and V in the ViT-style operator. This is probably not ideal.
1. The subdomains are not periodic, so padding/continuation is always necessary.
2. There are harmonics that will cannot be captured well on a subdomain grid, since it's relatively coarse.
     - one potential workaround is to upsample subdomains, but that would be kind of bad...?

TL;DR "think of how to compute Q, K, and V." There's a gazillion variants of neural operators that are potentially usable.

**Comment by arthurfeeney** [comment]
Another idea that may be fun is to use the local attention inside each subdomain to compute qkv.
Probably kinda overkill, but potentially cool 😎

**Comment by arthurfeeney** [comment]
Using spectral conv actually seems to do really bad on different resolutions when the subdomains are small. On the training resolution it does well. Tried different number of modes.
